sigstore/transparency.py

```diff
@@ -76,6 +76,13 @@ class LogEntry:
     The base64-encoded Signed Entry Timestamp (SET) for this log entry.
     """
 
+    # NOTE: After Rekor bundles (provided by `--rekor-bundle`) are removed, this will no longer be
+    # necessary.
+    _from_rekor_bundle: bool
```

Comment:
This isn't great, but we'll be able to remove it soon when we get rid of `--rekor-bundle` so I'm not too worried about it.

The only problem I see is that even though this field is marked as internal, you still need to provide a value for it if you construct one yourself. But I don't think users of the API have a good reason to be doing that anyway.

Reply:
This also outputs a `RuntimeWarning` from Pydantic at runtime:

> `RuntimeWarning: fields may not start with an underscore, ignoring "_from_rekor_bundle"`

Reply:
Yeah, I think we can do a little better than this API (both the private fiel;d and the manual initialization). I'll look into it.

Reply:
Hmm, maybe we can just remove `--rekor-bundle` outright? We never "actually" stabilized it (we've always emitted a warning said that it's temporary and will be removed in an upcoming release), and we never integrated it into other tooling (like the GH Action) or other expectations (e.g. CPython doesn't list it).

Thoughts @di?

Reply:
Fine with me!

Reply:
Cool! I'll just yank all of that out, then, and save us from even having to think about this 🙂